AI-154: Styled Search Builder Page

mihaelmladenov opened the pull request:
# Description:

### This PR styles:
* Search builder page (/civicrm/contact/search/builder)

### This PR changes the CSS code without affecting the styling for:
* Dashboard page (/civicrm)

# Screenshots:

## Search builder page (/civicrm/contact/search/builder)

### Before:
![before](https://user-images.githubusercontent.com/16223739/32595870-31af5950-c53a-11e7-831a-9b96ee967847.png)

### After:
![after](https://user-images.githubusercontent.com/16223739/32595880-37df2922-c53a-11e7-99ea-a27b5bf87fa2.png)

## Dashboard page (/civicrm)

### Before:
![dashboard-before](https://user-images.githubusercontent.com/16223739/32597316-f61d0e82-c53e-11e7-9f38-2b86b9cb463b.png)

### After:
![dashboard-after](https://user-images.githubusercontent.com/16223739/32597323-f906c1d8-c53e-11e7-8a50-0824c620b639.png)

# Technical Details:
Apart from styling the search builder page, this PR:
* introduces a new mixin called "section-title" to be used for styling section titles (like the "Include contacts where" title in the screenshots above);
* replaces the section title CSS of the dashboard getting started boxes with the new mixin.

AkA84 commented on scss/civicrm/search/pages/_search-builder.scss:
```diff
@@ -0,0 +1,32 @@
+.CRM_Contact_Form_Search_Builder {
+  #map-field tr td {
```
always nest
```scss
#map-field {
  tr {
    td {
      padding
    }
  }
}

AkA84 commented on scss/civicrm/search/pages/_search-builder.scss:
```diff
@@ -0,0 +1,32 @@
+    }
+  }
+
+  #map-field .crm-search-builder-add-row td.form-item {
```
you already have the `#map-field` selector above, don't redefine it

AkA84 commented on scss/civicrm/search/pages/_search-builder.scss:
```diff
@@ -0,0 +1,32 @@
+  }
+
+  #crm-submit-buttons {
+    padding: 25px 20px;
```
this looks a very generic styling ("apply paddings to the submit buttons"), is this really necessary? don't we have this style already somewhere else?

mihaelmladenov commented:
@AkA84 please find below my comments:
1. Nesting of CSS rules has been fixed.
2. Combining CSS selectors has been fixed.
3. I have added to the description screenshots from the dashboard, where we are using the same "section-title". I have not found it yet at another place in CiviCRM, but since it's already used several times, I believe it's better to have a mixin for that.
4. I searched and did not find a general padding for these buttons elsewhere. On most of the other pages, the buttons are in a block together with the fields (inputs/selects/etc) above them and the whole wrapper has this padding. In our case, the content above spans from left to right and only the buttons need to have the padding on the left and right.
5. I have fixed the commit history.

jamienovick commented:
@mihaelmladenov does it also work for /civicrm/dashboard?reset=1? there was an issue with this before

mihaelmladenov commented:
@jamienovick Yes, I've changed the section titles in the dashboard page also.
